assets/src/edit-story/panels/panel/handle.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * External dependencies
+ */
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	useRef,
+	useState,
+	useCallback,
+	useLayoutEffect,
+} from '@wordpress/element';
+
+const Handle = styled.button`
+	background: transparent;
+	border: 0;
+	padding: 0;
+	position: absolute;
+	top: 0;
+	left: 50%;
+	margin-left: -60px;
+	width: 120px;
+	height: 20px;
+	display: flex;
+	flex-direction: column;
+	justify-content: flex-start;
+	align-items: center;
+	cursor: row-resize;
+`;
+
+const Bar = styled.div`
+	margin-top: 4px;
+	background-color: black;
+	width: 32px;
+	height: 4px;
+	border-radius: 2px;
+`;
+
+function DragHandle( { handleHeightChange } ) {
+	const handle = useRef();
+	const lastPosition = useRef();
+	const [ isDragging, setIsDragging ] = useState( false );
```

Comment:
Should this state be here or should be now be moved to the panel provider?

Reply:
I feel it belongs here. It's a local state only relevant inside this component.